lib/std/math/signbit.zig

```diff
@@ -2,61 +2,25 @@ const std = @import("../std.zig");
 const math = std.math;
 const expect = std.testing.expect;
 
-/// Returns whether x is negative or negative 0.
+/// Returns whether x is a negative number.
 pub fn signbit(x: anytype) bool {
     const T = @TypeOf(x);
-    return switch (T) {
-        f16 => signbit16(x),
-        f32 => signbit32(x),
-        f64 => signbit64(x),
-        f128 => signbit128(x),
-        else => @compileError("signbit not implemented for " ++ @typeName(T)),
-    };
-}
-
-fn signbit16(x: f16) bool {
-    const bits = @bitCast(u16, x);
-    return bits >> 15 != 0;
-}
-
-fn signbit32(x: f32) bool {
-    const bits = @bitCast(u32, x);
-    return bits >> 31 != 0;
-}
-
-fn signbit64(x: f64) bool {
-    const bits = @bitCast(u64, x);
-    return bits >> 63 != 0;
-}
-
-fn signbit128(x: f128) bool {
-    const bits = @bitCast(u128, x);
-    return bits >> 127 != 0;
+    const TBits = std.meta.Int(.unsigned, @bitSizeOf(T));
+    if (@typeInfo(T) != .Float) {
+        @compileError("copysign not implemented for " ++ @typeName(T));
```

Comment:
LGTM, only thing is this error message mismatches the name of the function.
```suggestion
        @compileError("signbit not implemented for " ++ @typeName(T));
```

Reply:
Thanks!